File: Pi/src/main/java/com/example/pi/entities/commande.java
package com.example.pi.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.Set;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level= AccessLevel.PRIVATE)
@Entity
public class commande {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    Long commandeID;
    @ManyToOne
    private user user;
    private String shippingAddress; // Shipping address

    private String status; // Order status (e.g., "pending", "shipped", etc.)

    private String paymentStatus; // Payment status (e.g., "paid", "pending", etc.)

    private String paymentMethod; // Payment method (e.g., "credit card", "paypal")

    private String shippingMethod; // Shipping method (e.g., "standard", "express")

    private double shippingCost;



    @OneToMany(cascade = CascadeType.ALL)
    private Set<item> items;
}

File: Front/src/app/models/commande.ts
import { Cart } from './cart';
import { User } from './user';

export interface Commande {
  commandeID: number | null;
  cart: Cart | null; 
  shippingAddress: string;
  status: string;
  paymentStatus: string;
  paymentMethod: string;
  shippingMethod: string;
  shippingCost: number;
}


export { Cart };
